scripts/checkstyle.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+import re
+import sys
+import os
+import argparse
+from pathlib import Path
+from typing import List, Dict, Any # compat
+
+class c:
+    RED = '\033[91m'
+    GREEN = '\033[92m'
+    YELLOW = '\033[93m'
+    BLUE = '\033[94m'
+    MAGENTA = '\033[95m'
+    CYAN = '\033[96m'
+    RESET = '\033[0m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+
+# Before: printf '%s\\n' '^w^'
+# After: printf '%s\n' '^w^'
+def noDoubleBackslashFixer(line: str, rule: Dict[str, str], m: re.Match[str]) -> str:
```

Comment:
Not saying we should block this PR because of this, more curiosity than anything, is there a way for us to write some python tests and have them inline in this same file?

Reply:
For instance, I came across a tool last week which would be a good candidate to generate the regex with: https://github.com/pemistahl/grex
>automatically generating a single regular expression from user-provided test cases

Reply:
If I am understanding your first question correctly, each fixer function isn't inline because Python's lambas are expression-only and they aren't able to span multiple lines - a tradegy. I worry grex won't be general enough, but if it works well, then it might be useful.

Reply:
My first question was just about where you can put tests for Python code. I am not familiar with the patterns. Ideally, we wouldn't populate the `scripts/` directory with all the file structure of a Python library. So can we write tests in the `scripts/checkstyle.py` file? If we can, can we capture each of the cases found in this particular PR as test cases. You said you already had to update the Regex to capture another case in `command-current.bash`, so there is already an example of us finding an edge case. Since we want to guard against regressions, test cases would be valuable.

Reply:
Yeah good idea - test case would be nice. I can just add a flag like `--internal-test-regex` and have an extra array for positive and negative cases within the `rules` array. Is it OK if I hold off on doing that until I add another rule (probably one that matches [this](https://github.com/asdf-vm/asdf/pull/1349/commits/16836ef766813968b17a784fdee789a1c0af86c8) changeset)

Reply:
About doing the right patterns for the tests, I chose not to do any convention from any particular ecosystem because putting everything in a single file is important for simplicity.

Reply:
Sounds good to me. Thanks